### DISCUSSION TITLE
Shell functions vs playbooks

### DISCUSSION THREAD

**Discussion by ResistanceIsUseless** [pry0cc/axiom#362] [Ideas]
I created an issue to clean up the shell functions since .zshrc is getting pretty long. Maybe this is a better place to start the discussion on that because I feel like playbooks aren't really used(unless i missed something recently). I could see implementing more functions and oneliners to playbooks after we come up with guidelines on what should be a function vs a playbook and how playbooks are used.

My suggestion is we create a separate folder for functions (similar to fish) and functions are only use for OS Utility related commands such as starting and stopping a service or a container. Playbooks will be used for a single command or a one-liner with multiple steps so you don't have remember everything.

**Comment by 0xtavian** [comment]
@ResistanceIsUseless I love the idea of cleaning up .zshrc and you are correct about playbooks, they arent used and probably need some attention. Another important part that needs clean up are the packer provisioners. Ideally I image chain of image builders https://medium.com/swlh/chaining-machine-image-builds-with-packer-b6fd99e35049 but thats subject. Anything you can do would be much appreciated. LMK how I can help. I've just been a little busy lately so havent been able to dedicate as much time.

**Reply by ResistanceIsUseless** [reply]
What was the plan for playbooks? Where they intended as a type of script or more of a cheatsheet? I like the idea of both and could see add a lot of [these](https://github.com/dwisiswant0/awesome-oneliner-bugbounty) being added after cleaning them up.

I agree the provisioners need some work. Are you thinking we should clean up the profiles and provisioners and use chaining more? I think that makes sense. From my perspective Axiom has shifted to a more mass task platform as opposed to a quickly spun up cloud tooling platform. Obviously there is no reason we can't have both and i personally use it for the latter more.  I've always liked the idea of more specific roles for image builds. It would be great to start adding more custom purpose images and even infrastructure(multiple images or managed clusters) that serve a purpose. Here are a few ideas.. 

-Main Provisioner 
--Additional Tools added to main Image (More pentesting related tools such as metasploit, shell managing, etc)
-SSRF/XXE/XSS OOB Testing Image (Interactsh, web service with php files,XXE payloads, etc)
-VPN Server 
-C2 Server (covenant, metasploit, etc)
-CTF Tools Image
-Proxy Image (Burp, ZAP, VNC)
-Logging/Analytics Image (ElasticSearch, Splunk, other log reviewing tools)
-intrigue-core Image (https://github.com/intrigueio/intrigue-core)
-Vulnerable Machines (Build main provisioner image and spin up something like JuiceShop as well)

I could keep going but you get the point. Axiom is great for creating workers but I don't want to lose the idea of using it for Infrastructure as Code. I can throw together most of these images but I would like some outline on if we should use profiles or a whole new provisioner file.

Also, I thought i recall you mentioning you and @pry0cc split off the wiki into a new repo? Is that correct? because I cant find it or is that still in progress. Feel free to message me on Discord if its easier to chat in real time. -StaticBunny

**Reply by 0xtavian** [reply]
https://github.com/pry0cc/axiom/issues/376#issuecomment-874299878 more issues with Packer provisioner. Just leaving it here as a note
